physicsnemo/models/domino/model.py

```diff
@@ -199,49 +237,66 @@ def __init__(self, input_filters, model_parameters):
         self.activation = F.relu
         self.batch_norm = False
 
-    def forward(self, x):
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """
+        Process geometry information through the 3D CNN network.
+
+        The network follows an encoder-decoder architecture with skip connections:
+        1. Downsampling path (encoder) with three levels of max pooling
+        2. Processing loop in the bottleneck
+        3. Upsampling path (decoder) with skip connections from the encoder
+
+        Args:
+            x: Input tensor containing grid-represented geometry of shape
+               (batch_size, input_filters, nx, ny, nz)
+
+        Returns:
+            Processed geometry features of shape (batch_size, 1, nx, ny, nz)
+        """
         # Encoder
         x0 = x
+        x = self.conv1(x)
         if self.batch_norm:
-            x = self.activation(self.conv_bn1(self.conv1(x)))
-        else:
-            x = self.activation(self.conv1(x))
+            x = self.conv_bn1(x)
+        x = self.activation(x)
         x = self.max_pool(x)
+
         x1 = x
+        x = self.conv2(x)
         if self.batch_norm:
-            x = self.activation(self.conv_bn2(self.conv2(x)))
-        else:
-            x = self.activation((self.conv2(x)))
+            x = self.conv_bn2(x)
+        x = self.activation(x)
         x = self.max_pool(x)
 
         x2 = x
+        x = self.conv3(x)
         if self.batch_norm:
-            x = self.activation(self.conv_bn3(self.conv2(x)))
```

Comment:
Note: this fixes an important functionality bug: in the 3rd level of the downsampling, conv2 was accidentally used twice, and conv3 was never used (in the batch_norm branch).